src/renderer/renderElement.ts

```diff
@@ -114,31 +116,39 @@ const elementWithCanvasCache = new WeakMap<
   ExcalidrawElementWithCanvas
 >();
 
+const shapeCache = new WeakMap<
+  ExcalidrawElement,
+  Drawable | Drawable[] | null
+>();
+
+export function getShapeForElement(element: ExcalidrawElement) {
```

Comment:
should this populate the `shapeCache`? right now, it's hard to know when the `shape` is populated.